planner/core/task.go

```diff
@@ -1021,9 +1021,20 @@ func (p *PhysicalProjection) GetCost(count float64) float64 {
 }
 
 func (p *PhysicalProjection) attach2Task(tasks ...task) task {
+	if cop, ok := tasks[0].(*copTask); ok {
+		if len(cop.rootTaskConds) == 0 && cop.getStoreType() == kv.TiFlash && expression.CanExprsPushDown(p.ctx.GetSessionVars().StmtCtx, p.Exprs, p.ctx.GetClient(), cop.getStoreType()) {
+			cop = attachPlan2Task(p, cop).(*copTask)
+			projConcurrency := p.ctx.GetSessionVars().DistSQLScanConcurrency()
+			if projConcurrency <= 0 {
+				projConcurrency = 1
+			}
+			cop.addCost(p.GetCost(tasks[0].count() / float64(projConcurrency)))
```

Comment:
Seems in `finishCopTask`, the cop task's cost will be divided by `p.ctx.GetSessionVars().DistSQLScanConcurrency()`, so the project's cost will be divided twice?

Reply:
it will return directly, not enter `finishCopTask `.

Reply:
I will be finished anyway. because the last task will finally become root task

Reply:
ok